src/main.rs

```diff
@@ -43,6 +156,28 @@ fn panic(panic: &core::panic::PanicInfo) -> ! {
     arch::oops(&pp)
```

Comment:
i think we should add a line to `arch::oops` like 

```rust
    #[cfg(test)]
    qemu_exit(QemuExitCode::Failed);
```
since the current oops handler will display the oops screen and then hang so the user can _read_ the oops screen. this means that if a test fails by panicking or by triggering a CPU fault, rather than exiting and reporting that the test failed, we will hang forever (which we especially don't want in CI). if we are going to fail tests using the Rust `assert` macro, we definitely need to exit QEMU in the panic handler.

alternatively, we could put `#[cfg(not(test))]` on the call to `arch::oops` here, and have the panic handler exit qemu when running tests, but that means that CPU exceptions would still not result in normal test failures...

Reply:
Yeah, I wanted to do this, but didn't in the first version, as your oops handler hadn't been written yet, and I didn't want to make my changes conflict :-)